apps/src/code-studio/progressRedux.js

```diff
@@ -75,11 +75,12 @@ export default function reducer(state = initialState, action) {
         }
 
         const id = level.uid || bestResultLevelId(level.ids, newLevelProgress);
+        let nextLevel = level;
         if (action.peerReviewsPerformed && stage.flex_category === 'Peer Review') {
-          Object.assign(level, action.peerReviewsPerformed[index]);
+          nextLevel = Object.assign({}, level, action.peerReviewsPerformed[index]);
```

Comment:
would it be more bullet proof to use object spread operators? Since you can't accidentally mutate an object with it?